### ISSUE TITLE
Support subgraph studio subgraphs

### ISSUE THREAD

**Issue by geleeroyale** [Giveth/giveth-dapps-v2#2836]
In order to keep the DApp running, we should need to our studio-deployed subgraph as soon as possible. The hosted version that we are using now will not be supported indefinitely and is actually **past its announced date of support**.

We also need this on the backend - the relevant issue is being created. Please notify devops once the changes can be merged, so we can have a seamless transition!

**Comment by mhmdksh** [comment]
Thanks @geleeroyale for opening this.

@MoeNick @aminlatifi Please note that this issue is directly responsible for all the GivEconomy Hick-ups we are facing. And there is nothing we can do from our end as Kay mentioned that the hosted version is not supported anymore from Graph themselves.

In other words, until this is finally closed, we'll keep getting giveconomy problems and Giveth will look like "It's Not Working......"

**Comment by mhmdksh** [comment]
@aminlatifi Yes sure, you can find them commented in the config files. Or double check the notion page that Kay worked on earlier. Here: https://www.notion.so/giveth/Subgraph-Studio-8a497caa2d6a43cb871cf8919f8e8bbd?pvs=4

**Comment by geleeroyale** [comment]
We created all the endpoints and the API key some time ago and shared it with @developers. Here you go:

https://www.notion.so/giveth/Subgraph-Studio-8a497caa2d6a43cb871cf8919f8e8bbd?pvs=4

**Comment by aminlatifi** [comment]
Thanks @geleeroyale and @mhmdksh

**Comment by aminlatifi** [comment]
@MoeNick Let's plan this issue for the next week. Subgraph endpoints URLs must be replaced with new ones mentioned in the docs

**Comment by aminlatifi** [comment]
@geleeroyale @mhmdksh Just a question, Do the endpoints support localhost origin? we may need that support in the testing

**Comment by geleeroyale** [comment]
We created all the endpoints and the API key some time ago and shared it with @developers. Here you go:

https://www.notion.so/giveth/Subgraph-Studio-8a497caa2d6a43cb871cf8919f8e8bbd?pvs=4

**Comment by RamRamez** [comment]
> We created all the endpoints and the API key some time ago and shared it with @Developers. Here you go:
> 
> https://www.notion.so/giveth/Subgraph-Studio-8a497caa2d6a43cb871cf8919f8e8bbd?pvs=4
> 
> Edit: We are happy to further assist this process

I don't have access to this notion.
@geleeroyale @mhmdksh Can you give me access? I requested with my gmail

**Comment by geleeroyale** [comment]
Sorry - access has been granted!

**Comment by RamRamez** [comment]
@geleeroyale Is it possible to test the studio api locally/localhost?

![image](https://github.com/Giveth/giveth-dapps-v2/assets/73441857/6036b8c5-8946-443a-8ec5-9b0d966081fd)

**Comment by RamRamez** [comment]
@geleeroyale I see staging URLs are already set in the dapp, but they are different from your notion URLs. Shall I update them too?

**Comment by geleeroyale** [comment]
@RamRamez Sorry - I was not able to make an exception for localhost. I removed the giveth domain. Let's not forget to add it back in when we are done.

I had an error in the Notion where I specified `development.ts` for production where it should be `production.ts` - The values in development.ts should be alright, the ones for production.ts should be updated according to the Notion IMO.

Happy to test this with you tomorrow @RamRamez - I'll ping you

**Comment by RamRamez** [comment]
I tested the giveconomy and it's working fine.

@geleeroyale I was thinking we can assign a subdomain for my branch (like "studio.giveth.io") for testers to test, then you can add *.giveth.io back to the subgraph config.

**Comment by divine-comedian** [comment]
@geleeroyale is this issue still relevant for us?

**Comment by mhmdksh** [comment]
> @geleeroyale is this issue still relevant for us?

@divine-comedian Yes this is still very relevant

**Comment by divine-comedian** [comment]
@mhmdksh @geleeroyale - what else needs to happen for this issue?
